pkg/minishift/update/update.go

```diff
@@ -264,12 +267,34 @@ func extractBinary(downloadedArchivePath, archiveDir string) (string, error) {
 			return "", err
 		}
 
-		binaryPath = filepath.Join(extract, "minishift")
+		binaryPath, err = minishiftBinaryPath(extract, "minishift")
+		if err != nil {
+			return "", err
+		}
 	}
 
 	return binaryPath, nil
 }
 
+// minishitBinaryPath take extracted Path and os specific binary name and return
+// absolute path of the binary
+func minishiftBinaryPath(extractPath, binaryName string) (string, error) {
+	var binaryPath string
+	var err error
+	fileList := []string{}
+	err = filepath.Walk(extractPath, func(path string, f os.FileInfo, err error) error {
+		fileList = append(fileList, path)
+		return nil
+	})
+	for _, fileName := range fileList {
+		if strings.Contains(fileName, binaryName) {
```

Comment:
I am not quite sure what @budhrg means with _login_, but afaict you walk the tree and append everything to a slice. Then you iterate the slice to see whether you got a match. Why not do everything in one go in the anonymous function passed to file path.Walk? Just check each entry whether it is a match. If so record the entry. After file path.Walk you just check whether you found something. 

Up to you what you want to do.